src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,142 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        T item;
+        Node<T> next;
+        Node<T> prev;
+
+        public Node(T value, Node<T> next, Node<T> prev) {
+            this.item = value;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    public MyLinkedList() {
+        first = null;
+        last = null;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode;
+        if (isEmpty()) {
+            newNode = new Node<>(value, null, null);
+            first = newNode;
+            last = newNode;
+        } else {
+            newNode = new Node<>(value, null, last);
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(index + " index is out of bounds");
+        }
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> oldNode = findNodeByIndex(index);
+            Node<T> prev = oldNode.prev;
+            Node<T> newNode = new Node(value, oldNode, prev);
+            oldNode.prev = newNode;
+            if (prev == null) {
+                first = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T item : list) {
+            add(item);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        exceptionCheck(index);
+        Node<T> node = findNodeByIndex(index);
+        T replacedValue = node.item;
+        node.item = value;
+        return replacedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        exceptionCheck(index);
```

Comment:
this call is redundant, cause you use this check inside of findNodeByIndex

Reply:
removed